Marketplace/whiteboard/markerEntityScript.js

```diff
@@ -0,0 +1,429 @@
+
+    }
+
+    // subscribe to channel 
+    var MarkerTip = function() {
+        _this = this;
+        _this.whiteboards = [];
+        _this.colors = [];
+        _this.STROKE_NAME = "hifi_polyline_markerStroke";
+        _this.WHITEBOARD_SURFACE_NAME = "Whiteboard - Drawing Surface";
+        _this.MARKER_COLOR_NAME = "hifi-whiteboardPaint";
+    };
+
+    function handleColorMessage(channel, message, senderID) {
+        if (channel === "Ink-Color") {
+            var auxMessage = JSON.parse(message);
+            var data = utils.getEntityUserData(_this.entityID);
+            data.markerColor = auxMessage.markerColor;
+            var serverID = getServerID();
+            // RPC - Call server change marker ink 
+            Entities.callEntityServerMethod(serverID, 'serverSetEntityData', [_this.entityID, JSON.stringify(data)]);
+        }
+    }
+
+    function getServerID() {
+        if (_this.currentWhiteboard !== null) {
+            return Entities.getEntityProperties(_this.currentWhiteboard, "parentID").parentID;
+        }
+        if (_this.whiteboards.length >= 1) {
+            return Entities.getEntityProperties(_this.whiteboards[0], "parentID").parentID;
+        }
+    }
+
+    Messages.subscribe("Ink-Color");
+    Messages.messageReceived.connect(handleColorMessage);
+
+    MarkerTip.prototype = {
+        preload: function(entityID) {
+            this.entityID = entityID;
+
+            Controller.mouseMoveEvent.connect(_this.mouseMoveEvent);
+            Controller.mouseReleaseEvent.connect(_this.mouseReleaseEvent);
+        },
+        startNearGrab: function() {
+            _this.whiteboards = [];
+            _this.colors = [];
+            var userData = utils.getEntityUserData(_this.entityID);
+            _this.markerColor = userData.markerColor;
+
+            var markerProps = Entities.getEntityProperties(_this.entityID);
+            _this.DRAW_ON_BOARD_DISTANCE = markerProps.dimensions.z / 2;
+            var markerPosition = markerProps.position;
+            var results = Entities.findEntities(markerPosition, WHITEBOARD_SEARCH_RADIUS);
+            results.forEach(function(entity) {
+                var entityName = Entities.getEntityProperties(entity, "name").name;
+                if (entityName === _this.WHITEBOARD_SURFACE_NAME) {
+                    _this.whiteboards.push(entity);
+                } else if (entityName === _this.MARKER_COLOR_NAME) {
+                    _this.colors.push(entity);
+                }
+            });
+            
+            var serverID = getServerID();
+
+            Entities.callEntityServerMethod(serverID, 
+                'spawnMarker', 
+                [_this.entityID, JSON.stringify(markerProps.name), JSON.stringify(_this.markerColor)]
+            );
+        },
+        releaseGrab: function() {
+            _this.resetStroke();
+        },
+        continueNearGrab: function(ID, paramsArray) {
```

Comment:
please write function parameters in camelcase, `id` or `entityID` would be valid

Reply:
got it!